docs/ci.md

```diff
@@ -0,0 +1,152 @@
+---
+title: With CI
+description: "Test that your images build and your servers start"
+layout: docs
+---
+
+Once you've bought into container-based development,
+you'll want to set up CI to ensure your servers don't break.
+
+We think of this in two pieces:
+
+1) Run Tilt in CI to deploy to a single-use cluster.
+
+2) Run a single-use Kubernetes in CI that you can immediately throw away.
+
+Let's dig into examples of each one.
+
+## The `tilt ci` command
+
+The [`tilt ci` command](/cli/tilt_ci.html) is made for CI jobs.
+
+### How it Works
+
+Under the hood, `tilt ci`:
+
+1) Executes your Tiltfile.
+
+2) Runs all `local_resource` commands.
+
+3) Builds all images.
+
+4) Deploys all Kubernetes resources.
+
+5) Waits until all servers and other Kubernetes resources are healthy.
+
+`tilt ci` defaults to log-streaming mode. The web UI is still accessible,
+but the terminal will be a simple log stream.
+
+If any step fails, Tilt will exit immediately with an error code.
+
+If any resource crashes or looks like it will never succeed (e.g., a pod that
+takes too long to schedule), Tilt will also exit with an error code.
+
+Once all services are healthy, it will exit with status code 0.
+
+### Examples
+
+Most of our example projects use CircleCI to run `tilt ci`:
+
+<ul>
+  {% for page in site.data.examples %}
+     <li><a href="/{{page.href | escape}}#ci">{{page.title | escape}}</a></li>
+  {% endfor %}
+</ul>
+
+These each use [`ctlptl`](https://ctlptl.dev/) to set up a single-use
+cluster. But if you're considering other options, read on.
+
+## Single-use Kubernetes clusters
+
+If you're using Kubernetes for dev, you'll likely also want to use Kubernetes in CI.
+
+Tilt-team has this problem as well! Our integration test suite uses the latest
+version of Tilt to deploy real sample projects against a real cluster.  So we
+have a lot of experiences with best practices.
+
+Here are some options, with the pros and cons of each:
+
+### Easiest: Cluster on VM-based CI
+
+[`kind`](https://kind.sigs.k8s.io/) is currently the gold standard for running
```

Comment:
nbd but I like adding a little context -- consider some/all of the below
```suggestion
[`kind`](https://kind.sigs.k8s.io/) (Kubernetes in Docker, which runs a single-node Kubernetes cluster as a Docker container) is currently the gold standard for running
```

Reply:
added some detail. also it's no longer true that kind clusters are single-node!
https://pkg.go.dev/sigs.k8s.io/kind/pkg/apis/config/v1alpha4#Cluster

Reply:
🤯